lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -302,6 +305,10 @@ bool SslClientConnection::connectSocket() {
 
   _errorDetails.clear();
   _socket = _endpoint->connect(_connectTimeout, _requestTimeout);
+  long flags = fcntl(_socket.fileDescriptor, F_GETFL, 0);
```

Comment:
We should not call `fcntl` in case we don't need to. This will likely run a system call otherwise, which we would like to avoid if not needed.